docker/Dockerfile

```diff
@@ -0,0 +1,81 @@
+# Copyright 2023 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+# Builder image
+###############################################################################
+FROM redhat/ubi9 AS builder
+
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETPLATFORM
+ARG KUBESTELLAR_VERSION=v0.3.1
+
+WORKDIR /
+
+# install requirements
+RUN yum install -y wget
+
+# download kubectl binaries
+RUN wget -nv "https://dl.k8s.io/release/v1.25.3/bin/${TARGETPLATFORM}/kubectl" -O kubectl && \
```

Comment:
Why `wget` here and `curl` later?

It looks to me like `curl` could be used:

```console
mspreitz@mjs12 docker % docker run --rm -it redhat/ubi9 sh

sh-5.1# curl -L --no-progress-meter -o /tmp/k https://dl.k8s.io/release/v1.25.3/bin/linux/amd64/kubectl

sh-5.1# ls -l /tmp/k
-rw-r--r--. 1 root root 45015040 Jul 14 20:11 /tmp/k

sh-5.1# chmod a+x /tmp/k

sh-5.1# /tmp/k version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-12T10:57:26Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
The connection to the server localhost:8080 was refused - did you specify the right host or port?
```

Reply:
The only reason there is a two-stage build here --- and thus the only reason that podman is not sufficient --- is the choice to use wget in place of curl here, right?  I think it would better if the build were single stage.

Reply:
I don't think this is something that we should hold up the PR for.  I am ok with using 2 different methods in the same file if that is what the author is comfortable with.  Lets let this pass.